code/modules/mob/living/carbon/human/emote.dm

```diff
@@ -587,6 +617,22 @@ wink, yawn, swish, sway/wag, fastsway/qwag, stopsway/swag"}
 			for (var/mob/O in (hearers(src.loc, null) | get_mobs_in_view(world.view,src)))
 				O.show_message(message, m_type)
 
+	if (ipc_act)
+		var/old_screen
+
+		if(!(ipc_act in ipc_emote_list))	return //checks the emote is a valid one
+
+		if(src.h_style in ipc_emote_list)	return //checks that current screen isn't an emote
+
+		old_screen = h_style
+		h_style = ipc_act
+		update_emote()
+
+		sleep(30) //time the emote displays for

```

Comment:
Not an issue, persay, but in this case, since you don't _need_ to sleep the entire proc, using spawn is preferred.

You'd use sleep if you want to hold up the execution.